### PR TITLE
[Enhancement] Handle returned status (DLA part)

### PR DESCRIPTION
**Pull request by letian-jiang**
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 40 / 66 (60.61%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/sink/iceberg_table_sink_operator.cpp|0|1|00.00%|[101]|
|:large_blue_circle:|src/formats/parquet/stored_column_reader.cpp|0|1|00.00%|[634]|
|:large_blue_circle:|src/exec/orc_scanner.cpp|0|1|00.00%|[86]|
|:large_blue_circle:|src/runtime/jdbc_driver_manager.cpp|0|4|00.00%|[61, 93, 103, 137]|
|:large_blue_circle:|src/formats/parquet/column_reader.h|0|2|00.00%|[123, 124]|
|:large_blue_circle:|src/exec/parquet_writer.cpp|0|4|00.00%|[37, 40, 81, 88]|
|:large_blue_circle:|src/connector/jdbc_connector.cpp|0|1|00.00%|[94]|
|:large_blue_circle:|src/connector/mysql_connector.cpp|0|1|00.00%|[98]|
|:large_blue_circle:|src/connector/es_connector.cpp|0|1|00.00%|[223]|
|:large_blue_circle:|src/formats/parquet/file_reader.cpp|3|7|42.86%|[319, 323, 374, 378]|
|:large_blue_circle:|src/formats/orc/orc_chunk_reader.cpp|1|2|50.00%|[907]|
|:large_blue_circle:|src/exec/iceberg/iceberg_delete_builder.cpp|8|12|66.67%|[63, 64, 66, 67]|
|:large_blue_circle:|src/formats/parquet/column_reader.cpp|2|3|66.67%|[917]|
|:large_blue_circle:|src/exec/iceberg/iceberg_delete_file_iterator.cpp|4|4|100.00%|[]|
|:large_blue_circle:|src/util/arrow/row_batch.cpp|4|4|100.00%|[]|
|:large_blue_circle:|src/formats/parquet/column_chunk_reader.cpp|4|4|100.00%|[]|
|:large_blue_circle:|src/formats/parquet/file_writer.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/exec/hdfs_scanner.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/exec/connector_scan_node.cpp|2|2|100.00%|[]|
|:large_blue_circle:|src/formats/parquet/page_reader.cpp|3|3|100.00%|[]|
|:large_blue_circle:|src/exec/hdfs_scanner_orc.cpp|5|5|100.00%|[]|
|:large_blue_circle:|src/exec/parquet_scanner.cpp|2|2|100.00%|[]|

**Review by dirtysalt — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 41 / 68 (60.29%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/sink/iceberg_table_sink_operator.cpp|0|1|00.00%|[101]|
|:large_blue_circle:|src/formats/parquet/stored_column_reader.cpp|0|1|00.00%|[634]|
|:large_blue_circle:|src/exec/orc_scanner.cpp|0|1|00.00%|[86]|
|:large_blue_circle:|src/runtime/jdbc_driver_manager.cpp|0|4|00.00%|[61, 93, 103, 137]|
|:large_blue_circle:|src/formats/parquet/column_reader.h|0|2|00.00%|[123, 124]|
|:large_blue_circle:|src/exec/parquet_writer.cpp|0|4|00.00%|[37, 40, 81, 88]|
|:large_blue_circle:|src/connector/jdbc_connector.cpp|0|1|00.00%|[94]|
|:large_blue_circle:|src/connector/mysql_connector.cpp|0|1|00.00%|[98]|
|:large_blue_circle:|src/connector/es_connector.cpp|0|1|00.00%|[223]|
|:large_blue_circle:|src/formats/parquet/file_reader.cpp|3|7|42.86%|[319, 323, 374, 378]|
|:large_blue_circle:|src/formats/orc/orc_chunk_reader.cpp|1|2|50.00%|[907]|
|:large_blue_circle:|src/formats/parquet/column_reader.cpp|2|4|50.00%|[917, 918]|
|:large_blue_circle:|src/exec/iceberg/iceberg_delete_builder.cpp|8|12|66.67%|[63, 64, 66, 67]|
|:large_blue_circle:|src/exec/iceberg/iceberg_delete_file_iterator.cpp|4|4|100.00%|[]|
|:large_blue_circle:|src/util/arrow/row_batch.cpp|4|4|100.00%|[]|
|:large_blue_circle:|src/formats/parquet/column_chunk_reader.cpp|4|4|100.00%|[]|
|:large_blue_circle:|src/formats/parquet/file_writer.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/exec/hdfs_scanner.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/exec/connector_scan_node.cpp|2|2|100.00%|[]|
|:large_blue_circle:|src/formats/parquet/page_reader.cpp|3|3|100.00%|[]|
|:large_blue_circle:|src/exec/hdfs_scanner_orc.cpp|6|6|100.00%|[]|
|:large_blue_circle:|src/exec/parquet_scanner.cpp|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 41 / 69 (59.42%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/sink/iceberg_table_sink_operator.cpp|0|1|00.00%|[101]|
|:large_blue_circle:|src/formats/parquet/stored_column_reader.cpp|0|1|00.00%|[634]|
|:large_blue_circle:|src/exec/orc_scanner.cpp|0|1|00.00%|[86]|
|:large_blue_circle:|src/runtime/jdbc_driver_manager.cpp|0|4|00.00%|[61, 93, 103, 137]|
|:large_blue_circle:|src/formats/parquet/column_reader.h|0|3|00.00%|[123, 124, 126]|
|:large_blue_circle:|src/exec/parquet_writer.cpp|0|4|00.00%|[37, 40, 81, 88]|
|:large_blue_circle:|src/connector/jdbc_connector.cpp|0|1|00.00%|[94]|
|:large_blue_circle:|src/connector/mysql_connector.cpp|0|1|00.00%|[98]|
|:large_blue_circle:|src/connector/es_connector.cpp|0|1|00.00%|[223]|
|:large_blue_circle:|src/formats/parquet/file_reader.cpp|3|7|42.86%|[319, 323, 374, 378]|
|:large_blue_circle:|src/formats/orc/orc_chunk_reader.cpp|1|2|50.00%|[907]|
|:large_blue_circle:|src/formats/parquet/column_reader.cpp|2|4|50.00%|[917, 918]|
|:large_blue_circle:|src/exec/iceberg/iceberg_delete_builder.cpp|8|12|66.67%|[63, 64, 66, 67]|
|:large_blue_circle:|src/exec/iceberg/iceberg_delete_file_iterator.cpp|4|4|100.00%|[]|
|:large_blue_circle:|src/util/arrow/row_batch.cpp|4|4|100.00%|[]|
|:large_blue_circle:|src/formats/parquet/column_chunk_reader.cpp|4|4|100.00%|[]|
|:large_blue_circle:|src/formats/parquet/file_writer.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/exec/hdfs_scanner.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/exec/connector_scan_node.cpp|2|2|100.00%|[]|
|:large_blue_circle:|src/formats/parquet/page_reader.cpp|3|3|100.00%|[]|
|:large_blue_circle:|src/exec/hdfs_scanner_orc.cpp|6|6|100.00%|[]|
|:large_blue_circle:|src/exec/parquet_scanner.cpp|2|2|100.00%|[]|

**Review by Smith-Cruise — APPROVED** [review]

File: be/src/exec/parquet_writer.cpp [hunk 1]
```diff
@@ -34,10 +34,10 @@ RollingAsyncParquetWriter::RollingAsyncParquetWriter(
           _commit_func(std::move(commit_func)),
           _state(state),
           _driver_id(driver_id) {
-    init_rolling_writer();
+    init();
```
[the remark this answers is not in the document]
Comment (letian-jiang) [reply be/src/exec/parquet_writer.cpp]: removed

File: be/src/exec/iceberg/iceberg_delete_builder.cpp [hunk 2]
```diff
@@ -52,6 +59,12 @@ Status ParquetPositionDeleteBuilder::build(const std::string& timezone, const st
             }
         }
     }
+
+    // eof is expected, otherwise propagate error
+    if (!status.is_end_of_file()) {
+        LOG(WARNING) << status;
```
Comment (jaogoy) [inline be/src/exec/iceberg/iceberg_delete_builder.cpp:65]: The warning message is some simple, how could the users know the problem?

Comment (letian-jiang) [reply be/src/exec/iceberg/iceberg_delete_builder.cpp:65]: this is to print warning log to ease debugging, not for user

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by dirtysalt — APPROVED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.2

**Pull request merged**